### DISCUSSION TITLE
Best practice for hostname handling in multi-tenant deployment

### DISCUSSION THREAD

**Discussion by franck102** [keycloak/keycloak#20477] [Q&A]
Hi all, we are deploying Keycloak in a multi-tenant context; we are creating dedicated <tenant>.<our domain>.io domains for each tenant, and I am unsure about how to handle those hostnames with keycloak?

I can use 
```
hostname-strict: true // 
Disables dynamically resolving the hostname from request headers.
Should always be set to true in production, unless proxy verifies the Host header.
```

but then there is a domain name mismatch and I am not sure about its consequences:

```
https://**some-tenant.our-domain.io**/auth/realms/master/.well-known/openid-configuration
{
  "issuer": "https://www.our-domain.io/auth/realms/master",
  "authorization_endpoint": "https://**www.our-domain.io**/auth/realms/master/protocol/openid-connect/auth",
  "token_endpoint": "https://www.our-domain.io/auth/realms/master/protocol/openid-connect/token",
```
www.our-domain.io will be routed correctly, however won't the hostname mismatch break the authorization flows?
We basically use oidc-client-ts on the client to authenticate, and our backend services verify the bearer tokens using the Quarkus OIDC client.

Thanks!

**Comment by franck102** [comment]
Answering my own question, below is a configuration that works. As explained in the Keycloak documentation, do make sure that your proxy validates the Host and X-Forwarded* headers:

 ```
                    KC_PROXY: "edge",
                    KC_HTTP_ENABLED: "true",
                    KC_HOSTNAME: "www.our-domain.io",
                    KC_HTTP_RELATIVE_PATH: "/auth",
                    // Always use hostname above for frontend endpoints
                    KC_HOSTNAME_STRICT: "false",
                    // Resolve host from incoming request for backend endpoints since they use VPC endpoints
                    KC_HOSTNAME_STRICT_BACKCHANNEL: "false",
                    KC_HTTP_PORT: 8080,

```

**Reply by wissam-khair** [reply]
Hi .. can you please share with us your proxy forwarding config?
Thank you
